syntax = "proto3";
enum EObjectType {
    OT_NONE   = 0;  //无类别
    OT_PLAYER = 1;  //玩家
    OT_ROBOT  = 2;  //机器人
    OT_MONSTER= 3;  //怪
    OT_PET    = 4;  //宠物
    OT_PARTNER= 5;  //伙伴
    OT_SUMMON = 6;  //招唤物
}

enum EActionType {
    AT_NONE      = 0;  //无
    AT_IDLE      = 1;  //静
    AT_WALK      = 2;  //走
    AT_RUN       = 3;  //跑
    AT_FLY       = 4;  //飞
    AT_DANCE1    = 5;  //跳舞动作1;
    AT_DANCE2    = 6;  //跳舞动作2;
    AT_DANCE3    = 7;  //跳舞动作3;
    AT_DANCE4    = 8;  //跳舞动作4;
}

enum EActionStatue {
    AS_NONE         = 0;          //无动作
    AS_HURT         = 0x1;        //伤害
    AS_START        = 0x2;        //开始动作
    AS_END          = 0xfffffff;  //技能动作开始
}
    
enum ECopyType {
    CPT_NONE     = 0;        //无阵营
    CPT_LOGIN    = 1;        //登陆副本
    CPT_SELROLE  = 2;        //选角副本
    CPT_CITY     = 3;        //主城
    CPT_MAIN     = 4;        //主线副本
    CPT_WORLD    = 5;        //大世界
    
    CPT_END = 100;
}

enum EObjectStatus
{
    EOS_NONE            = 0;//无
    EOS_DEAD            = 1;//死亡
    EOS_IDLE            = 2;//静止
    EOS_WALK            = 3;//走
    EOS_RUN             = 4;//跑
    EOS_FLY             = 5;//飞
    EOS_NOT_MOVE        = 6;//不能移动
    EOS_NOT_CAST        = 7;//不能放技能
    EOS_NOT_BEHURT      = 8;//不能被攻击
    EOS_NOT_BECONTROL   = 9;//无法被控制
    EOS_STEALTH         = 10;//隐身
    EOS_BLIND           = 11;//瞎的
};


enum EItemType
{
    EIT_NONE         = 0;
    EIT_EQUIP        = 1;
    EIT_ACTIONDRUG   = 2;
    EIT_MAT          = 3;
    EIT_BOX          = 4;
    EIT_KEY          = 5;
    EIT_CHIP         = 6;
    EIT_GEM          = 7;
    EIT_FASHION      = 8;
    EIT_MONEY        = 9;
    EIT_ACTION       = 10;
    EIT_EXP          = 11;
    EIT_RUNE         = 12;
    EIT_PETSOUL      = 13; 
    EIT_TASK         = 14;
    EIT_DRUG         = 15;
    EIT_PET          = 16;
    EIT_PARTNER      = 17;
}

enum EBagType
{
    EBT_NONE    = 0;
    EBT_ITEM    = 1;
    EBT_GEM     = 2;
    EBT_FASHION = 3;
    EBT_CHIP    = 4;
    EBT_RUNE    = 5;
    EBT_TASK    = 6;
    EBT_SOUL    = 7;
    EBT_CARD    = 8;  //英雄卡
}

enum EFunctionType{
    FUNC_BEGIN_ID  = 0;  //!功能ID开始标记
    
    FUNC_END_ID = 150; //!功能ID结束标记
}


enum EAttrID{
    HP            = 0;  //生命值
    MP            = 1;  //魔法值
    HP_MAX        = 2;  //最大生命值
    MP_MAX        = 3;  //最大魔法值
    SPEED         = 4;  //移动速度
    ATTACK        = 5;  //攻击
    ELEMENT       = 6;  //元素伤害
    MAGIC_DEF     = 7;  //魔防
    PHYSIC_DEF    = 8;  //物防
    HIT_RATE      = 9; //命中
    DODGE         = 10; //闪避
    MORE_HURT     = 11; //伤害加深
    LESS_HURT     = 12; //伤害减免
    CRIT_HIT      = 13; //暴击
    CRIT_DEF      = 14; //抗暴
    CRIT_HURT     = 15; //暴伤
    ATTR_NUM  　　= 16; //属性个数
}

enum EShip
{
    TYPE_NONE    =  0;
    TYPE_ALLY    =  1;
    TYPE_ENEMY   =  2;
    TYPE_NEUTRAL =  3;
}


enum EBuffType{
    BFT_NONE        = 0; 
}

enum EChatChannel{
    CHL_NONE        = 0; 
    CHL_WORLD       = 1;//世界聊天
    CHL_PRIVATE     = 2;//私聊
    CHL_GUILD       = 3;//公会
    CHL_GM          = 4;//GM命令
}


enum EEquipPos{
    EEP_NONE        = 0; 
    EEP_HELMET      = 1; 
    EEP_NECKLACE    = 2; 
    EEP_ARMOR       = 3; 
    EEP_SHOES       = 4; 
    EEP_WRIST       = 5; 
    EEP_RING        = 6; 
    EEP_TALISMAN    = 7; 
    EEP_WEAPON      = 8; 
}

enum ETaskStatus{
    ETS_INIT        = 0; 
    ETS_ACCEPT      = 1; 
    ETS_FINISH      = 2; 
    ETS_COMMIT      = 3;
}

